src/ui/StarRating/StarRating.ts

```diff
@@ -0,0 +1,125 @@
+import 'styling/_StarRating';
+import { $$ } from '../../utils/Dom';
+import { Utils } from '../../utils/Utils';
+import { l } from '../../strings/Strings';
+import { Component } from '../Base/Component';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { exportGlobally } from '../../GlobalExports';
+import { IQueryResult } from '../../rest/QueryResult';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions, IFieldOption } from '../Base/ComponentOptions';
+import { Initialization } from '../Base/Initialization';
+
+const DEFAULT_SCALE = 5;
+
+export interface IStarRatingOptions {
+  ratingField?: IFieldOption;
+  numberOfRatingsField?: IFieldOption;
+  ratingScale?: number;
+}
+/**
+ * The `StarRating` component renders a 5-star rating widget for commerce results.
+ *
+ * This component is a result template component (see [Result Templates](https://developers.coveo.com/x/aIGfAQ)).
+ */
+export class StarRating extends Component {
+  static ID = 'StarRating';
+  private rating: number;
+  private numberOfRatingsField: number;
+
+  static doExport = () => {
+    exportGlobally({
+      StarRating: StarRating
+    });
+  };
+
+  static options: IStarRatingOptions = {
+    /**
+     * Specifies the rating to be displayed as stars.
+     *
+     * Specifying a value for this parameter is required in order for the StarRating component to work.
+     */
+    ratingField: ComponentOptions.buildFieldOption({ defaultValue: '@rating', required: true }),
+
+    /**
+     * Specifies the value to be displayed in the label indicating the quantity of ratings.
+     *
+     * If unspecified, no label is shown. If a value of `0` is provided, a `(No Ratings)` label is displayed instead.
+     */
+    numberOfRatingsField: ComponentOptions.buildFieldOption({ required: false }),
+
+    /**
+     * Specifies the scale on which ratings are to be applied
+     *
+     * Default value is `5`
+     */
+    ratingScale: ComponentOptions.buildNumberOption({ defaultValue: DEFAULT_SCALE, min: 1, max: 100000 })
```

Comment:
Mmm... sadly, I think TypeScript won't be able to resolve the value of this constant.

Would it be okay to set it to 5 instead (only here, in the buildNumberOption call's argument)?

Reply:
Seems to work though. If the component is instantiated without specifying this option scale is set to 5.

Reply:
ah sorry I wrote TypeScript and I actually meant TypeDoc haha...

So what I mean is that TypeDOC ( :D ) probably won't be able to resolve it, and therefore we'll see something like `Default: DEFAULT_SCALE` in the generated reference site.

Reply:
Ah got it! Will do.